File: src/main/java/com/durhack/sharpshot/nodes/routing/ACRotateNode.java
package com.durhack.sharpshot.nodes.routing;

import com.durhack.sharpshot.Bullet;
import com.durhack.sharpshot.Direction;
import com.durhack.sharpshot.INode;
import javafx.scene.Node;
import javafx.scene.control.Label;
import javafx.scene.layout.StackPane;
import javafx.scene.paint.Color;
import javafx.scene.shape.Rectangle;
import org.jetbrains.annotations.NotNull;

import java.math.BigInteger;
import java.util.HashMap;
import java.util.Map;

public class ACRotateNode implements INode {
    private Direction dir = Direction.UP;

    @Override
    public @NotNull Direction getRotation() {
        return dir;
    }

    @Override
    public String toString() {
        return "Rotate Anticlockwise";
    }

    @Override
    public void rotateClockwise() {
        dir = Direction.clockwiseOf(dir);
    }

    @Override
    public @NotNull Map<Direction, BigInteger> run(@NotNull Bullet bullet) {
        HashMap<Direction, BigInteger> map = new HashMap<>();
        map.put(bullet.getDirection().antiClockwise(), bullet.getValue());
        return map;
    }

    @Override
    public @NotNull Node toGraphic() {
        Rectangle rectangle = new Rectangle(32.0, 32.0, Color.PALEVIOLETRED);
        Label label = new Label("ACW");
        return new StackPane(rectangle, label);
    }

    public void reset() {}
}

File: src/main/java/com/durhack/sharpshot/nodes/routing/ACRotateNode.kt
package com.durhack.sharpshot.nodes.routing

import com.durhack.sharpshot.logic.Bullet
import com.durhack.sharpshot.GRID_SIZE
import com.durhack.sharpshot.nodes.INode
import javafx.scene.Node
import javafx.scene.control.Label
import javafx.scene.layout.StackPane
import javafx.scene.paint.Color
import javafx.scene.shape.Rectangle

class ACRotateNode : INode() {
    override fun run(bullet: Bullet) = mapOf(bullet.direction.antiClockwise to bullet.value)

    override fun graphic(): Node {
        val rectangle = Rectangle(GRID_SIZE.toDouble(), GRID_SIZE.toDouble(), Color.PALEVIOLETRED)
        val label = Label("ACW")
        return StackPane(rectangle, label)
    }

    override val type = "rotate anticlockwise"

    override fun reset() {}
    override val tooltip = "Rotates incoming bullets anticlockwise"
}
